quickwit/quickwit-proto/protos/quickwit/common.proto

```diff
@@ -0,0 +1,42 @@
+// Copyright (C) 2022 Quickwit, Inc.
+//
+// Quickwit is offered under the AGPL v3.0 and as commercial software.
+// For commercial licensing, contact us at hello@quickwit.io.
+//
+// AGPL:
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as
+// published by the Free Software Foundation, either version 3 of the
+// License, or (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program. If not, see <http://www.gnu.org/licenses/>.
+
+syntax = "proto3";
+
+package quickwit_common;
+
+message SetQuery {
+  // List of terms to search.
+  repeated Term terms = 1;
+  // Field to search in.
+  string field_name = 2;
+  // Union of tags a split must have ot be searched.
+  repeated string tags = 3;
+}
+
+message Term {
```

Comment:
overkill?
Down the line users are using text in the query parser. We probably want to have them enter text here too, and parse that text the way we parse it in the query parser.

Reply:
things here don't go through the query parser, they get converted [here](https://github.com/quickwit-oss/quickwit/blob/e4268d41cf4657a4c326e11047bb9c27f0c2eb94/quickwit/quickwit-doc-mapper/src/query_builder.rs#L102-L116) once we have a schema to know what type these terms are supposed to be